internal/cli/runner_uninstall.go

```diff
@@ -0,0 +1,171 @@
+package cli
+
+import (
+	"sort"
+	"strings"
+
+	"github.com/hashicorp/waypoint-plugin-sdk/terminal"
+	"github.com/hashicorp/waypoint/internal/clierrors"
+	"github.com/hashicorp/waypoint/internal/pkg/flag"
+	"github.com/hashicorp/waypoint/internal/runnerinstall"
+	pb "github.com/hashicorp/waypoint/pkg/server/gen"
+	"github.com/posener/complete"
+)
+
+type RunnerUninstallCommand struct {
+	*baseCommand
+
+	platform     string
+	mode         string
+	serverUrl    string
+	serverCookie string
+	id           string
+}
+
+func (c *RunnerUninstallCommand) AutocompleteArgs() complete.Predictor {
+	return complete.PredictNothing
+}
+
+func (c *RunnerUninstallCommand) AutocompleteFlags() complete.Flags {
+	return c.Flags().Completions()
+}
+
+func (c *RunnerUninstallCommand) Flags() *flag.Sets {
+	return c.flagSet(0, func(set *flag.Sets) {
+		f := set.NewSet("Command Options")
+
+		f.StringVar(&flag.StringVar{
+			Name:   "platform",
+			Usage:  "Platform to uninstall the Waypoint runner from.",
+			Target: &c.platform,
+		})
+
+		f.StringVar(&flag.StringVar{
+			Name:   "server-addr",
+			Usage:  "Address of the Waypoint server.",
+			EnvVar: "WAYPOINT_ADDR",
+			Target: &c.serverUrl,
+		})
+
+		f.StringVar(&flag.StringVar{
+			Name:   "id",
+			Usage:  "ID of the Waypoint runner.",
+			Target: &c.id,
+		})
+
+		// Add platforms in alphabetical order. A consistent order is important for repeatable doc generation.
+		i := 0
+		sortedPlatformNames := make([]string, len(runnerinstall.Platforms))
+		for name := range runnerinstall.Platforms {
+			sortedPlatformNames[i] = name
+			i++
+		}
+		sort.Strings(sortedPlatformNames)
+
+		for _, name := range sortedPlatformNames {
+			platform := runnerinstall.Platforms[name]
+			platformSet := set.NewSet(name + " Options")
+			platform.UninstallFlags(platformSet)
+		}
+	})
+}
+
+func (c *RunnerUninstallCommand) Help() string {
+	return formatHelp(`
+Usage: waypoint runner uninstall [options]
+
+  Uninstall a Waypoint runner from server given a platform and Waypoint runner
+  id. The platform should be specified as kubernetes, nomad, ecs, or docker.
+
+  This will forget the runner on the server and remove any of the resources
+  created by a runner installation.
+` + c.Flags().Help())
+}
+
+func (c *RunnerUninstallCommand) Run(args []string) int {
+	ctx := c.Ctx
+	log := c.Log.Named("uninstall")
+	defer c.Close()
+
+	// Initialize. If we fail, we just exit since Init handles the UI.
+	if err := c.Init(
+		WithArgs(args),
+		WithFlags(c.Flags()),
+		WithNoLocalServer(), // no auth in local mode
+		WithNoConfig(),
+	); err != nil {
+		return 1
+	}
+
+	if c.platform == "" {
+		c.ui.Output(
+			"The -platform flag is required.",
```

Comment:
Instead of requiring a platform flag. we should load the CLI context and read the platform out of there. I believe that's how the server uninstall works today! If there is no platform defined in the CLI context, we can fail if the platform flag wasn't specified.

Reply:
Great minds! 😄